message.go

```diff
@@ -247,13 +251,19 @@ func checkMessage(m Message) error {
 	}
 
 	if len(m.Data) < 4 {
-		return errShortErrorMessage
+		return &OpError{
+			Op:  "receive",
+			Err: errShortErrorMessage,
+		}
 	}
 
 	if c := nlenc.Int32(m.Data[0:4]); c != success {
 		// Error code is a negative integer, convert it into
 		// an OS-specific system call error
-		return newError(-1 * int(c))
+		return &OpError{
```

Comment:
I'm having a difficult time deciding if this change is needed, and whether or not the internal newError should do os.NewSyscallError.

These are errors derived from netlink messages, not from the system calls used to perform them.  What's the correct answe here?  Should we have another type like MessageError or something to differentiate?

Reply:
ISTM like the system call itself was successful, but the action the message was meant to take was not. Therefore, it feels like this is not the right place for a `SyscallError`, since `SyscallError` records an error from a specific call.

Where would the hypothetical `MessageError` fit in? Would it be returned instead of `OpError` on some code paths? Or would it be returned in the `Err` field of an `OpError`?

`newError` returns a `syscall.Errno`, and there is not much more context to it than that. This time, the _caller_ has the context, because they know what high level operation they are doing. How about the following lightweight convention: if `.Err` is an `*os.SyscallError`, then the caller knows that it is a system call that failed; if `.Err` is a naked `syscall.Errno`, then the operation itself has failed. If the caller _really_ cares about this beyond what `IsNotExist` or similar functions can offer, then we can simply document this convention.

Reply:
I see what you mean!  I will revert this, but I will document OpError.Err to explain the convention.

Reply:
I'm confused. Is there anything to revert here? The change seems fine as-is.

Reply:
I think I got confused as well!  I'll proceed according to the convention you mentioned above.

Reply:
Okay, I updated the docs and did a new commit with nltest changes too to check the behavior!

Reply:
Excellent. This behavior LGTM now.